src/ruby/yast/debugger.rb

```diff
@@ -0,0 +1,157 @@
+
+require "yast"
+
+module Yast
+  class Debugger
+    class << self
+      include Yast::Logger
+      include Yast::UIShortcuts
+
+      # Start the Ruby debugger. It handles the current UI mode and displays
+      # an user request if the debugger front-end needs to be started manually.
+      # @param [Boolean] remote if set to true the server is accesible from network.
+      #   By default the debugger can connect only from the local machine, not from
+      #   the network. If you need remote debugging then enable it.
+      #   WARNING: There is no authentication, everybody can connect to
+      #   the debugger! Use only in a trusted network as this is actually
+      #   a backdoor to the system! For secure connection use SSH and start
+      #   the debugger locally after connecting via SSH.
+      # @param [Fixnum] port the port number where the debugger server will
+      #   listen to
+      # @param [Boolean] start_client autostart the debugger client
+      #   (ignored in remote debugging)
+      # @example Start the debugger with default settings:
+      #   require "yast/debugger"
+      #   Yast::Debugger.start
+      # @example When using the debugger temporary you can use just simple:
+      #   require "byebug"
+      #   byebug
+      def start(remote: false, port: 3344, start_client: true)
+        begin
+          require "byebug"
+          require "byebug/core"
+        rescue LoadError
+          # catch loading error, the debugger is optional (might not be present)
+          Yast.import "Report"
+          Report.Error(format("Cannot load the Ruby debugger.\n" \
+            "Make sure '%s' Ruby gem is installed.") % "byebug")
+          return
+        end
+
+        popup = false
+
+        # do not start the server if it is already running
+        if Byebug.started?
+          log.warn "The debugger is already running at port #{Byebug.actual_port}"
+          log.warn "Skipping the server setup"
+        else
+          Yast.import "UI"
+          wait = false
+          if UI.TextMode || remote || !start_client
+            # in textmode or in remote mode ask the user to start
+            # the debugger client manually
+            UI.OpenDialog(Label(debugger_message(remote, port)))
+            popup = true
+          else
+            # in GUI open an xterm session with the debugger
+            start_gui_session(port)
+            # wait a bit to get the server ready (to avoid "Broken pipe" error)
+            # FIXME: looks like a race condition inside byebug itself...
+            wait = true
+          end
+
+          # start the server and wait for connection
+          start_server(remote, port, wait)
```

Comment:
wait parameter looks a bit weird, as I interpret it as wait for connection. and it is also boolean parameter, so maybe named parameter will be nicer.